File: Proj9_1/app/src/main/java/com/proto/proj9_1/MainActivity.java
package com.proto.proj9_1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.Path;
import android.graphics.Rect;
import android.graphics.RectF;
import android.graphics.drawable.shapes.OvalShape;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.LinearLayout;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
//        setContentView(new MyGraphicView(this), paramsG);

        LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        LinearLayout lay1 = new LinearLayout(this);
        lay1.setOrientation(LinearLayout.VERTICAL);
        setContentView(lay1, params);

        LinearLayout.LayoutParams btnparams1 = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        Button btn1 = new Button(this);
        btn1.setText("그림판으로 넘어가기");
        lay1.addView(btn1, btnparams1);
        lay1.addView(new MyGraphicView(this), btnparams1);

        btn1.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view){
                Intent intent = new Intent(getApplicationContext(), p9_1.class);
                startActivity(intent);
            }
        });
    }

    private static class MyGraphicView extends View {
        public MyGraphicView(Context context){
            super(context);
        }

        @Override
        protected void onDraw(Canvas canvas){
            super.onDraw(canvas);
            Paint paint =new Paint();
            paint.setAntiAlias(true);
            paint.setColor(Color.GREEN);
            canvas.drawLine(10, 10, 300, 10, paint);

            paint.setColor(Color.BLUE);
            paint.setStrokeWidth(5);
            canvas.drawLine(10, 30, 300, 30, paint);

            paint.setColor(Color.RED);
            paint.setStrokeWidth(0);

            paint.setStyle(Paint.Style.FILL);
            Rect rect1 = new Rect(10, 40, 10+100, 50+100);
            canvas.drawRect(rect1, paint);

            paint.setStyle(Paint.Style.STROKE);
            Rect rect2 = new Rect(130, 50, 130+100, 50+100);
            canvas.drawRect(rect2, paint);

            RectF rect3 = new RectF(250, 50, 250+100, 50+100);
            canvas.drawRoundRect(rect3, 20, 20, paint);

            canvas.drawCircle(60, 220, 50, paint);

            paint.setStrokeWidth(5);
            Path path1 = new Path();
            path1.moveTo(10, 290);
            path1.lineTo(10 + 50, 290 + 50);
            path1.lineTo(10 + 100, 290 + 50);
            path1.lineTo(10 + 200, 290);
            canvas.drawPath(path1, paint);

            paint.setStrokeWidth(0);
            paint.setTextSize(30);
            canvas.drawText("안드로이드", 10, 390, paint);

            paint.setStrokeWidth(5);
            paint.setTextSize(30);
            canvas.drawText("안드로이드", 10, 490, paint);

            paint.setColor(Color.BLACK);
            paint.setStyle(Paint.Style.FILL);
            Rect rect4 = new Rect(10, 550, 10+100, 550+20);
            canvas.drawRect(rect4, paint);

            RectF rect5 = new RectF(10, 600, 10+100, 600+20);
            canvas.drawRoundRect(rect5, 20, 20, paint);

            RectF rect6 = new RectF(10, 650, 10+100, 650+20);
            canvas.drawOval(rect6, paint);

            float startAngle = 70f, sweepAngle = 90f;

            RectF arc = new RectF(130, 650, 130+50, 650+50);
            canvas.drawArc(arc, startAngle, sweepAngle, true, paint);

            paint.setColor(Color.argb(255, 255, 0, 0 ));
            Rect rect7 = new Rect(250, 700, 250+200, 700+200);
            canvas.drawRect(rect7, paint);

            paint.setColor(Color.argb(150, 0, 0, 255));
            Rect rect8 = new Rect(200, 650, 200+200, 650+200);
            canvas.drawRect(rect8, paint);
        }

    }

}

File: Proj9_1/app/src/main/java/com/proto/proj9_1/MainActivity.kt
package com.proto.proj9_1

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.LinearLayout
import android.view.ViewGroup
import android.content.Intent
import android.graphics.*
import com.proto.proj9_1.p9_1
import android.view.View
import android.widget.Button

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //        setContentView(new MyGraphicView(this), paramsG);
        val params = LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        val lay1 = LinearLayout(this)
        lay1.orientation = LinearLayout.VERTICAL
        setContentView(lay1, params)
        val btnparams1 = LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        val btn1 = Button(this)
        btn1.text = "그림판으로 넘어가기"
        lay1.addView(btn1, btnparams1)
        lay1.addView(MyGraphicView(this), btnparams1)
        btn1.setOnClickListener {
            val intent = Intent(applicationContext, p9_1::class.java)
            startActivity(intent)
        }
    }

    private class MyGraphicView(context: Context?) : View(context) {
        override fun onDraw(canvas: Canvas) {
            super.onDraw(canvas)
            val paint = Paint()
            paint.isAntiAlias = true
            paint.color = Color.GREEN
            canvas.drawLine(10f, 10f, 300f, 10f, paint)
            paint.color = Color.BLUE
            paint.strokeWidth = 5f
            canvas.drawLine(10f, 30f, 300f, 30f, paint)
            paint.color = Color.RED
            paint.strokeWidth = 0f
            paint.style = Paint.Style.FILL
            val rect1 = Rect(10, 40, 10 + 100, 50 + 100)
            canvas.drawRect(rect1, paint)
            paint.style = Paint.Style.STROKE
            val rect2 = Rect(130, 50, 130 + 100, 50 + 100)
            canvas.drawRect(rect2, paint)
            val rect3 = RectF(250F, 50F, 250F + 100F, 50F + 100F)
            canvas.drawRoundRect(rect3, 20f, 20f, paint)
            canvas.drawCircle(60f, 220f, 50f, paint)
            paint.strokeWidth = 5f
            val path1 = Path()
            path1.moveTo(10f, 290f)
            path1.lineTo((10 + 50).toFloat(), (290 + 50).toFloat())
            path1.lineTo((10 + 100).toFloat(), (290 + 50).toFloat())
            path1.lineTo((10 + 200).toFloat(), 290f)
            canvas.drawPath(path1, paint)
            paint.strokeWidth = 0f
            paint.textSize = 30f
            canvas.drawText("안드로이드", 10f, 390f, paint)
            paint.strokeWidth = 5f
            paint.textSize = 30f
            canvas.drawText("안드로이드", 10f, 490f, paint)
            paint.color = Color.BLACK
            paint.style = Paint.Style.FILL
            val rect4 = Rect(10, 550, 10 + 100, 550 + 20)
            canvas.drawRect(rect4, paint)
            val rect5 = RectF(10F, 600F, 10F + 100F, 600F + 20F)
            canvas.drawRoundRect(rect5, 20f, 20f, paint)
            val rect6 = RectF(10F, 650F, 10F + 100F, 650F + 20F)
            canvas.drawOval(rect6, paint)
            val startAngle = 70f
            val sweepAngle = 90f
            val arc = RectF(130F, 650F, 130F + 50F, 650F + 50F)
            canvas.drawArc(arc, startAngle, sweepAngle, true, paint)
            paint.color = Color.argb(255, 255, 0, 0)
            val rect7 = Rect(250, 700, 250 + 200, 700 + 200)
            canvas.drawRect(rect7, paint)
            paint.color = Color.argb(150, 0, 0, 255)
            val rect8 = Rect(200, 650, 200 + 200, 650 + 200)
            canvas.drawRect(rect8, paint)
        }
    }
}